cmd/collector/app/span_processor.go

```diff
@@ -58,19 +66,24 @@ type queueItem struct {
 
 // NewSpanProcessor returns a SpanProcessor that preProcesses, filters, queues, sanitizes, and processes spans.
 func NewSpanProcessor(
-	spanWriter spanstore.Writer,
+	traceWriter tracestore.Writer,
 	additional []ProcessSpan,
 	opts ...Option,
 ) (processor.SpanProcessor, error) {
-	sp, err := newSpanProcessor(spanWriter, additional, opts...)
+	sp, err := newSpanProcessor(traceWriter, additional, opts...)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not create span processor: %w", err)
 	}
 
 	sp.queue.StartConsumers(sp.numWorkers, func(item queueItem) {
 		sp.processItemFromQueue(item)
 	})
 
+	err = sp.otelExporter.Start(context.Background(), sp.telset.Host)
```

Comment:
@yurishkuro what's the reason we're starting a new context here rather than propagating it from upstream?